bigtable/google/cloud/bigtable/row_data.py

```diff
@@ -194,8 +194,66 @@ class PartialRowsData(object):
 
     def __init__(self, response_iterator):
         self._response_iterator = response_iterator
+        self.generator = YieldRowsData(response_iterator)
+
         # Fully-processed rows, keyed by `row_key`
         self._rows = {}
+
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return False
+        return other._response_iterator == self._response_iterator
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    @property
+    def state(self):
+        """State machine state.
+
+        :rtype: str
+        :returns:  name of state corresponding to currrent row / chunk
+                   processing.
+        """
+        return self.generator.state
+
+    @property
+    def rows(self):
+        """Property returning all rows accumulated from the stream.
+
+        :rtype: dict
+        :returns: row_key -> :class:`PartialRowData`.
+        """
+        # NOTE: To avoid duplicating large objects, this is just the
+        #       mutable private data.
+        return self._rows
+
+    def consume_all(self, max_loops=None):
+        """Consume the streamed responses until there are no more.
+
+        :type max_loops: int
+        :param max_loops: (Optional) Maximum number of times to try to consume
+                          an additional ``ReadRowsResponse``. You can use this
+                          to avoid long wait times.
+        """
+        for row in self.generator.read_rows():
+            self._rows[row.row_key] = row
+
+
+class YieldRowsData(object):
+    """Convenience wrapper for consuming a ``ReadRows`` streaming response.
+
+    :type response_iterator: :class:`~google.cloud.exceptions.GrpcRendezvous`
+    :param response_iterator: A streaming iterator returned from a
+                              ``ReadRows`` request.
+    """
+    START = "Start"  # No responses yet processed.
```

Comment:
1. Please use single quotes (i.e. `'Start'`, `'New row'`, etc.)
1. Please leave an empty line between the docstring and the class members / constants